src/function/vector_cast_operations.cpp

```diff
@@ -21,6 +23,16 @@ scalar_exec_func VectorCastOperations::bindExecFunc(
                                             common::Types::dataTypeToString(targetTypeID) + ".");
         }
     }
+    case common::FLOAT: {
```

Comment:
You may do this after u add other numerical types.

Instead of doing double switch, you can extract the second switch as another function just as what u do in build_in_vector_operation